tiledb/sm/array_schema/dimension.cc

```diff
@@ -1591,16 +1598,15 @@ Status Dimension::check_tile_extent() const {
     case Datatype::TIME_AS:
       return check_tile_extent<int64_t>();
     default:
-      return LOG_STATUS(Status_DimensionError(
-          "Tile extent check failed; Invalid dimension domain type"));
+      throw DimensionException(
+          name() + " [check_tile_extent] Invalid dimension domain type");
```

Comment:
This is an example of what this looks like in practice:
```
Dimension: cols [check_tile_extent] Tile extent 5 exceeds range on dimension domain [1, 4]
```
I think it's a little confusing, and it doesn't really match our other error messages. My suggestion:
* Drop `[check_tile_extent]`. We generally don't include the function name in error messages.
* Say what the `name()` is instead of just prepending it to the start of the message. For example,  end with  `on dimension 'cols'`.